apps/wallet/src/ui/app/components/WalletListSelectItem.tsx

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { CheckFill16 } from '@mysten/icons';
+import { cx } from 'class-variance-authority';
+
+import { useMiddleEllipsis } from '../hooks';
+import { Text } from '../shared/text';
+
+export type WalletListSelectItemProps = {
+    address: string;
+    selected: boolean;
+};
+
+export function WalletListSelectItem({
+    address,
+    selected,
+}: WalletListSelectItemProps) {
+    const addressShort = useMiddleEllipsis(address);
```

Comment:
for addressShort can you use ` formatAddress` from mysten/sui.js

Reply:
I think useMiddleEllipsis is used almost everywhere, we should probably add a task to add a hook to format the address

Reply:
Added the ticket https://mysten.atlassian.net/browse/APPS-526